### PR TITLE
daemon/config: remove commonUnixBridgeConfig and CommonUnixConfig

### PR DESCRIPTION
**Pull request by thaJeztah**
These types were added as part of the "Solaris" support, because Solaris didn't support these options. Having these types added another layer of "abstraction" when reading the code, making the code harder to read, and more complicated to use.

Now that implementing support for Solaris has been abandoned (and removed), we can re-integrate these fields in the "unix" configuration.

See individual commits for details.

### CODE REVIEW

File: daemon/config/config_unix.go [hunk 1]
```diff
@@ -1,10 +1,12 @@
-// +build linux freebsd
+// +build !windows
```
Comment (AkihiroSuda) [inline daemon/config/config_unix.go:1]: Can we rename the file to _linux.go and remove the build tag?

I don’t think the code in this file works on non-Linux.

Comment (thaJeztah) [reply daemon/config/config_unix.go]: Yeah, makes sense; I thought "if it compiles, it compiles" but probably better to be explicit (as we for sure don't _test_, or _assume_ other unix-y environments for these)

Updated 👍

File: daemon/config/config_linux.go [hunk 2]
```diff
@@ -104,7 +160,7 @@ func (conf *Config) ValidatePlatformConfig() error {
 	return verifyDefaultCgroupNsMode(conf.CgroupNamespaceMode)
 }
 
-// IsRootless returns conf.Rootless
+// IsRootless returns conf.Rootless on Unix but false on Windows
```
Comment (thaJeztah) [inline daemon/config/config_linux.go:163]: Let me change this one as well

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
@tianon @cpuguy83 PTAL 🤗 (best reviewed by going through the commits)

**Review by tianon — APPROVED** [review]
This LGTM, although there has been some effort recently around FreeBSD -- I suppose these are split nicely enough to be easily reverted if that work happens to start landing before this code changes again. :sweat_smile:

**Comment by thaJeztah** [comment]
>  I suppose these are split nicely enough to be easily reverted if that work happens to start landing before this code changes again.

Yes, I had that in the back of my mind when making these changes, then considered "YAGNI"; if we need to specialise these types again, it will _probably_ be slightly different, and I thought it'd be better to make changes based on what we need _then_ instead of hoping it'll be the same as it is now.

**Comment by thaJeztah** [comment]
Let me get this one in

**Pull request merged**